devito/yask/transformer.py

```diff
@@ -130,49 +135,54 @@ def nary2binary(args, op):
         if function.is_Constant:
             if function not in mapper:
                 mapper[function] = yc_soln.new_grid(function.name, [])
-            return mapper[function].new_relative_grid_point([])
-        elif not function.is_Dimension:
+            return mapper[function].new_grid_point([])
+        elif function.is_Dimension:
+            if expr.is_Time:
+                return nfac.new_step_index(expr.name)
+            elif expr.is_Space:
+                return nfac.new_domain_index(expr.name)
+            else:
+                return nfac.new_misc_index(expr.name)
+        else:
             # A DSE-generated temporary, which must have already been
             # encountered as a LHS of a previous expression
             assert function in mapper
             return mapper[function]
     elif expr.is_Indexed:
+        # Create a YASK compiler grid if it's the first time we encounter a Function
         function = expr.base.function
         if function not in mapper:
-            if function.is_TimeFunction:
-                dimensions = [nfac.new_step_index(function.indices[0].name)]
-                dimensions += [nfac.new_domain_index(i.name)
-                               for i in function.indices[1:]]
-            else:
-                dimensions = [nfac.new_domain_index(i.name)
-                              for i in function.indices]
+            dimensions = [make_yask_ast(i, yc_soln, mapper) for i in function.indices]
             mapper[function] = yc_soln.new_grid(function.name, dimensions)
-        # Detect offset from dimension. E.g., in `[x+3,y+4]`, detect `[3,4]`
+        # Convert the Indexed into a YASK grid access
         indices = []
-        for i, j in zip(expr.indices, function.indices):
+        for i in expr.indices:
             if isinstance(i, LoweredDimension):
-                access = i.origin
+                indices.append(make_yask_ast(i.origin, yc_soln, mapper))
+            elif i.is_integer:
+                # A YASK misc dimension causes this
```

Comment:
Technically any dimension that we access explicitly by index, but I agree that those are most likely to be misc.

Reply:
right, gonna tweak this comment